perf_guide.md

```diff
@@ -67,17 +67,49 @@ addition is 2.7x slower than the one using vertical vector operations!
 
 ## Target-features
 
+Not all processors of a certain architecture will have SIMD processing units,
+and using a SIMD instruction when it's not supported will trigger a CPU exception.
+
+To allow building safe, portable programs, the Rust compiler will **not**, by default,
+generate any sort of vector instructions, unless it can statically determine
+they are supported (for example, on AMD64, SSE2 support is architecturally guaranteed).
+
 ### RUSTFLAGS
 
-TODO
+One of the easiest ways to benefit from SIMD is to allow the compiler
+to generate code using certain vector instruction extensions.
+
+The environment variable `RUSTFLAGS` can be used to pass options for code
+generation to the Rust compiler. These flags will affect **all** compiled crates.
+
+There are two flags which can be used to enable specific vector extensions:
+
+- `-C target-feature=<features>`: a comma-separated set of instruction extensions
+  to enable.
+  - **NOTE** (x86 specific): the `avx2` feature will _not_ enable `fma`, even though
+  all CPUs which support AVX2 also support FMA. You can enable this feature in addition to,
+  or separately from AVX2.
+
+<!-- TODO: list all target features for each platform -->
 
-Enabling features for the whole binary using `-C target-feature` and `-C target-cpu`, ...
+- `-C target-cpu=<cpu>`: the identifier of a CPU family / model for which to build
+  and optimize the code.
+  - The compiler will translate this into a list of target features.
+  - It will also cause the code generator to optimize the generated code
+  for that specific CPU model.
 
-Arch specific information: e.g. `-C target-feature=+avx2` does not enable `fma` 
+<!-- TODO: list all target CPUs for each platform -->
 
 ### Attribute
 
-TODO
+`RUSTFLAGS` is a powerful tool but its extent is too wide. In some cases it might
```

Comment:
I read this paragraph a couple of times and felt a bit weird. I don't have any good suggestions here :/ 

Maybe something like "Because `RUSTFLAGS` enable features for the whole binary, supporting different sets of hardware often requires shipping different binaries to avoid undefined behavior. This can be cumbersome, particularly if you cannot make many assumptions about the hardware a binary will run on. Rust's`#[target_feature]` attribute solves this problem by allowing you to embed machine code for different hardware with different target features into a single binary.".

Or something like that?